sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,280 @@
+import { DeepMocked, createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, getISODateOnlyString } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../../test/helpers";
+import { FullTimeMSFAAProcessResponseIntegrationScheduler } from "../msfaa-full-time-process-response-integration.scheduler";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  mockDownloadFiles,
+  MSFAA_FULL_TIME_RECEIVE_FILE_WITH_REACTIVATION_RECORD,
+  MSFAAStates,
+  saveFakeApplicationDisbursements,
+  saveFakeStudent,
+  createFakeMSFAANumber,
+} from "@sims/test-utils";
+import { THROW_AWAY_MSFAA_NUMBER } from "./msfaa-helper";
+import {
+  ApplicationStatus,
+  DisbursementScheduleStatus,
+  OfferingIntensity,
+} from "@sims/sims-db";
+import * as Client from "ssh2-sftp-client";
+import { Job } from "bull";
+import * as path from "path";
+import { MSFAA_FULL_TIME_MARRIED } from "./msfaa-full-time-process-integration.scheduler.models";
+import { saveMSFAATestInputsData } from "./msfaa-factory";
+import { In, IsNull } from "typeorm";
+
+describe(
+  describeProcessorRootTest(QueueNames.FullTimeMSFAAProcessResponseIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: FullTimeMSFAAProcessResponseIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+    let msfaaMocksDownloadFolder: string;
+
+    beforeAll(async () => {
+      msfaaMocksDownloadFolder = path.join(__dirname, "msfaa-receive-files");
+      // Set the ESDC response folder to the files mocks folders.
+      process.env.ESDC_RESPONSE_FOLDER = msfaaMocksDownloadFolder;
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      // Processor under test.
+      processor = app.get(FullTimeMSFAAProcessResponseIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Force any not signed MSFAA to be signed to ensure that new
+      // ones created will be the only ones available to be updated.
+      await db.msfaaNumber.update(
+        { dateSigned: IsNull() },
+        { dateSigned: getISODateOnlyString(new Date()) },
+      );
+      // Cancel any not canceled MSFAA.
+      await db.msfaaNumber.update(
+        { cancelledDate: IsNull() },
+        { cancelledDate: getISODateOnlyString(new Date()) },
+      );
+    });
+
+    it("Should reactivate a cancelled MSFAA when the same MSFAA is received in the response file.", async () => {
+      // Arrange
+      const student = await saveFakeStudent(db.dataSource);
```

Comment:
I do not see a valid reason to create a student here and assign the same to `saveMSFAATestInputsData` unless I am missing something.